litex/soc/cores/lifcllram.py

```diff
@@ -0,0 +1,56 @@
+# This file is Copyright (c) 2019 William D. Jones <thor0505@comcast.net>
+# This file is Copyright (c) 2019 Tim 'mithro' Ansell <me@mith.ro>
+# This file is Copyright (c) 2019 Florent Kermarrec <florent@enjoy-digital.fr>
+# License: BSD
+
+from migen import *
+from litex.soc.interconnect import wishbone
+
+kB = 1024
+
+"""
+Crosslink-NX family-specific Wishbone interface to the LRAM primitive.
+
+Currently this supports only Crosslink-NX/40 which has 2 LRAMs. Crosslink-NX/17
+has 5 LRAMs and so may be configured for larger RAM sizes.
+
+Note that this memory is dual port, but we only use a single port in this
+instantiation.
+"""
+
+class LIFCLLRAM(Module):
+    def __init__(self, width=32, size=64*kB):
+        self.bus = wishbone.Interface(width)
+        assert width in [32, 64]
+
+        if width == 32:
+            assert size in [64*kB, 128*kB]
+            depth_cascading = size//(64*kB)
+            width_cascading = 1
+        if width == 64:
+            assert size in [128*kB]
+            depth_cascading = size//(64*kB)
```

Comment:
Shouldn't this be 128kB because you're using double the blocks with the double width?

Reply:
You're right: I messed up arithmetic in a few places.